src/System.Private.CoreLib/shared/System/Number.Parsing.cs

```diff
@@ -1393,110 +1339,217 @@ internal static unsafe bool TryParseUInt64(ReadOnlySpan<char> s, NumberStyles st
             if (IsWhite(num))
             {
                 if ((styles & NumberStyles.AllowTrailingWhite) == 0) goto FalseExit;
-                for (index++; index < source.Length; index++)
+                for (index++; index < value.Length; index++)
                 {
-                    if (!IsWhite(source[index])) break;
+                    if (!IsWhite(value[index])) break;
                 }
-                if ((uint)index >= (uint)source.Length) goto DoneAtEndButPotentialOverflow;
+                if ((uint)index >= (uint)value.Length) goto DoneAtEndButPotentialOverflow;
             }
 
-            if (!TrailingZeros(source, index)) goto FalseExit;
+            if (!TrailingZeros(value, index)) goto FalseExit;
             
             goto DoneAtEndButPotentialOverflow;
         }
 
-        internal static unsafe decimal ParseDecimal(ReadOnlySpan<char> value, NumberStyles options, NumberFormatInfo numfmt)
+        /// <summary>Parses ulong limited to styles that make up NumberStyles.HexNumber.</summary>
+        private static bool TryParseUInt64HexNumberStyle(
+            ReadOnlySpan<char> value, NumberStyles styles, NumberFormatInfo info, out ulong result, ref bool failureIsOverflow)
+        {
+            Debug.Assert((styles & ~NumberStyles.HexNumber) == 0, "Only handles subsets of HexNumber format");
+            Debug.Assert(!failureIsOverflow, $"failureIsOverflow should have been initialized to false");
+
+            if ((uint)value.Length < 1) goto FalseExit;
+
+            bool overflow = false;
+            int index = 0;
+            int num = value[0];
+            int numValue = 0;
+
+            // Skip past any whitespace at the beginning.  
+            if ((styles & NumberStyles.AllowLeadingWhite) != 0 && IsWhite(num))
+            {
+                index++;
+                while (true)
```

Comment:
Can we change this to be like the do/while pattern below?
https://github.com/dotnet/coreclr/pull/18952/files#diff-daa56fa64da52c949170dca7347c4ce6R1389

Reply:
Sure